src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -422,6 +424,17 @@ private static NamedTypeSymbol ValidateBuilderType(SyntheticBoundNodeFactory F,
                     !property.IsStatic &&
                     (property.ParameterCount == 0))
                 {
+                    HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                    Conversion conversion = F.Compilation.Conversions.ClassifyConversionFromType(property.Type, returnType, ref useSiteDiagnostics);
+                    if (conversion.Kind == ConversionKind.NoConversion)
+                    {
+                        var badTaskProperty = new CSDiagnostic(
+                            new CSDiagnosticInfo(ErrorCode.ERR_BadAsyncMethodBuilderTaskProperty, builderType, returnType, property.Type),
+                            F.Syntax.Location);
+                        F.Diagnostics.Add(badTaskProperty);
+                        return null;
```

Comment:
I wasn't concerned about the risk of adding a specific error message, just the inconsistency of having a specific message for this one case. (We could have a similar specific message when checking the return type in `GetCustomCreateMethod` above for instance.)

Regarding error messages that refer to `Task` or `Task<T>`, we have #12621.